sdk/keccak-hasher/src/lib.rs

```diff
@@ -1,34 +1,45 @@
 //! Hashing with the [keccak] (SHA-3) hash function.
 //!
 //! [keccak]: https://keccak.team/keccak.html
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![cfg_attr(feature = "frozen-abi", feature(min_specialization))]
 
 #[cfg(feature = "borsh")]
 use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+#[cfg(any(feature = "sha3", not(target_os = "solana")))]
+use sha3::{Digest, Keccak256};
+pub use solana_hash::{ParseHashError, HASH_BYTES, MAX_BASE58_LEN};
 use {
-    sha3::{Digest, Keccak256},
     solana_sanitize::Sanitize,
-    std::{convert::TryFrom, fmt, str::FromStr},
-    thiserror::Error,
+    std::{fmt, str::FromStr},
```

Comment:
how about making this crate `no_std` like the hash crate? this can use `core::{fmt, str::FromStr}`. It can have an `std` feature which pretty much just enables the `std` feature on solana-hash